diff --git a/share/html/m/ticket/reply b/share/html/m/ticket/reply
index 00c8e2b..18587bc 100644
--- a/share/html/m/ticket/reply
+++ b/share/html/m/ticket/reply
@@ -81,6 +81,7 @@
     InUnits => $ARGS{'UpdateTimeWorked-TimeUnits'}||'minutes',
 &>
 </span></div>
+% $m->callback( %ARGS, CallbackName => 'AfterWorked', Ticket => $t );
 <input type="hidden" class="hidden" name="id" value="<%$t->Id%>" /><br />
 <div class="entry"><span class="label"><&|/l&>Update Type</&>:</span>
 <div class="value"><select name="UpdateType">
@@ -240,12 +241,17 @@ $m->comp( '/Elements/GnuPG/SignEncryptWidget:Process',
     TicketObj => $t,
 );
 
-if ( !$checks_failure && exists $ARGS{SubmitTicket} ) {
+my $skip_update = 0;
+$m->callback( CallbackName => 'BeforeUpdate', ARGSRef => \%ARGS, skip_update => \$skip_update,
+              checks_failure => $checks_failure, results => \@results, TicketObj => $t );
+
+if ( !$checks_failure && !$skip_update && exists $ARGS{SubmitTicket} ) {
     my $status = $m->comp('/Elements/GnuPG/SignEncryptWidget:Check',
         self      => $gnupg_widget,
         TicketObj => $t,
     );
     $checks_failure = 1 unless $status;
+    $m->callback( Ticket => $t, ARGSRef => \%ARGS, CallbackName => 'BeforeDisplay' );
     return $m->comp('/m/ticket/show', TicketObj => $t, %ARGS);
 }
 </%INIT>
diff --git a/share/html/m/_elements/raw_style b/share/html/m/_elements/raw_style
index 02c95b5..e19e2a6 100644
--- a/share/html/m/_elements/raw_style
+++ b/share/html/m/_elements/raw_style
@@ -153,7 +153,7 @@ hr.clear {
 }
 
 
-.label, .labeltop {
+.label, .labeltop, .cflabel {
     font-weight: normal;
 }
 .value { 
@@ -449,7 +449,7 @@ div.txn-content {
 
 }
 
-.label {
+.label, .cflabel {
     text-align: left;
     width: 10em;
     color: #666;
